### PR TITLE
Error handling, unsubscribe, bug fix

### PR DESCRIPTION
**Pull request by KirinaK**

### CODE REVIEW

File: src/app/todo-page/new-item/new-item.component.ts [hunk 1]
```diff
@@ -27,8 +26,8 @@ export class NewItemComponent implements OnInit {
   @Output() changeItem = new EventEmitter<Todo[]>();
   @Output() sortItem = new EventEmitter<Todo[]>();
 
-  constructor(private todoService: TodoItemService, private connectionService: ConnectionService) {
-    this.subscription = this.connectionService.todoData$.subscribe(item => this.editItem(item));
+  constructor(private connectionService: ConnectionService) {
+    this.checkData();
```
Comment (fasvald) [inline src/app/todo-page/new-item/new-item.component.ts]:
```
    this.checkData();
```

@KirinaK Why you don't call this inside ngOnInit?

**Review by fasvald — CHANGES_REQUESTED** [review]
Check comment

File: src/app/todo-page/new-item/new-item.component.ts (hunk 1 above)
Comment (KirinaK) [reply src/app/todo-page/new-item/new-item.component.ts]: Moved to ngOnInit()

**Review by fasvald — APPROVED** [review]

**Comment by fasvald** [comment]
@KirinaK feel free to merge.

**Pull request merged**